src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,94 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { useEffect, useState } from 'react';
+import { Post } from '../types/Post';
+import { client } from '../utils/fetchClient';
+
+type Props = {
+  selectedPost: Post
+  commentsFromServer: () => void
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  selectedPost,
+  commentsFromServer,
+}) => {
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [text, setText] = useState('');
+  const [isNameDanger, setIsNameDanger] = useState(false);
+  const [isEmailDanger, setIsEmailDanger] = useState(false);
+  const [isTextDanger, setIsTextDanger] = useState(false);
+  const [isLoadingComment, setIsLoadingComment] = useState(false);
+
+  const newCommentPost = async () => {
+    await client.post('/comments', {
+      postId: selectedPost.id,
+      name,
+      email,
+      body: text,
+    });
+  };
+
+  const submitHandler = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    if (name !== '' && email !== '' && text !== '') {
+      newCommentPost();
+      setIsLoadingComment(true);
+    }
+
+    if (name === '') {
```

Comment:
Would be better to check the trimmed length. Check it for every field
```suggestion
    if (name.trim().length) {
```